### ISSUE TITLE
Include a better explanation for NullPointExceptions

### ISSUE THREAD

**Issue by alvarovp27** [IDEA-Research-Group/dmn4spark#2]
DMNExecutor.scala, line 40-42:

```scala
  def getDecisionsResults(map: java.util.HashMap[String, AnyRef]): Seq[String] = {
    decisions.map(d => dmnEngine.evaluateDecisionTable(d, map).getFirstResult.getEntry(d.getKey).toString)
  }
```

NullPointExceptions may arise if the dmn engine fails to evaluate the decision table `d`. `evaluateDecisionTable` might return 0 solutions, hence, `getFirstResult` provokes a NullPointer. 

Solutions and possible root causes:

- Revise the DMN table with key d.getKey

Common issues include:
- The `decision_key` of required DMN tables (decisions) aren't correctly specified.
- The name, expression and/or variable of the inputs aren't correctly specified.
- The input data does not meet any rule.

In next versions, we should catch that NullPointerException and include this information.

**Comment by scott-neu-iw** [comment]
Álvaro,

I have been trying to get your library to work in a Databricks instance and keep getting a NullPointException.  Having this feature would be helpful in debugging the issue.  Are you planning on adding some better message handling to address this like the issue shows above?

Thank you

**Comment by alvarovp27** [comment]
Yes, I'll be releasing a new version very soon.

By the way, could you share more details about the NullPointerException you are having?

**Comment by scott-neu-iw** [comment]
Here is the exception & stack trace:

SparkException: Failed to execute user defined function(functions$$$Lambda$4059/1748059186: (struct<ProductId:int,FormularyId:int,PriorAuthorization:boolean>) => struct<_9FD242D6-F2C6-4381-BD16-8EAA6C6EBC00:string>)
Caused by: NullPointerException: 

org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 8.0 failed 4 times, most recent failure: Lost task 0.3 in stage 8.0 (TID 111, 10.103.249.10, executor 0): org.apache.spark.SparkException: Failed to execute user defined function(functions$$$Lambda$4059/1748059186: (struct<ProductId:int,FormularyId:int,PriorAuthorization:boolean>) => struct<_9FD242D6-F2C6-4381-BD16-8EAA6C6EBC00:string>)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.generate_doConsume_0$(Unknown Source)
	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:732)
	at org.apache.spark.sql.execution.collect.UnsafeRowBatchUtils$.encodeUnsafeRows(UnsafeRowBatchUtils.scala:80)
	at org.apache.spark.sql.execution.collect.Collector.$anonfun$processFunc$1(Collector.scala:187)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.doRunTask(Task.scala:144)
	at org.apache.spark.scheduler.Task.run(Task.scala:117)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$8(Executor.scala:677)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1581)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:680)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
	at es.us.idea.dmn4spark.dmn.DMNExecutor.$anonfun$getDecisionsResults$1(DMNExecutor.scala:41)
	at scala.collection.immutable.Stream.map(Stream.scala:418)
	at es.us.idea.dmn4spark.dmn.DMNExecutor.getDecisionsResults(DMNExecutor.scala:41)
	at es.us.idea.dmn4spark.dmn.DMNSparkEngine$$anon$1.call(DMNSparkEngine.scala:70)
	at es.us.idea.dmn4spark.dmn.DMNSparkEngine$$anon$1.call(DMNSparkEngine.scala:67)
	at org.apache.spark.sql.functions$.$anonfun$udf$58(functions.scala:4688)
	... 15 more

**Comment by alvarovp27** [comment]
Implemented in 2611bce83378b0f0fa1a0ab649342b710e6309de 
Included in release 1.1.2 52b659796659a1b207c514209ea80b2ebf0bd395

Now `NullPointerExceptions` produced when the Camunda DMN Engine fails to  evaluate a DMN table are handled, returning `null` such cases.

Additionally, if a DMN table produces more than one output, those results will be represented as a JSON string.

@scott-neu-iw hope it works for you. If you have any trouble, I kindly ask you to open a new Issue. Please take a look here: https://github.com/IDEA-Research-Group/dmn4spark#common-dmn-modelling-mistakes-and-workarounds
I included there a new section explaining three common cases which may cause a failure in the evaluation of a DMN model.

Thank you very much.
